core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -413,7 +413,7 @@ class LogSegment private[log] (val log: FileRecords,
   override def toString: String = "LogSegment(baseOffset=" + baseOffset +
     ", size=" + size +
     ", lastModifiedTime=" + lastModified +
-    ", largestTime=" + largestTimestamp +
+    ", largestRecordTimestamp=" + largestRecordTimestamp +
```

Comment:
Should we keep `largestTime` from LHS, and, print both `largestRecordTimestamp` and `largestTime` ?

Reply:
I'm ok with the change. I think it's better to reflect the underlying fields directly and redundant information just adds noise to the logs.